### PR TITLE
Rename packages: `code` -> `source_code`

### LINKED ISSUE #1126: [BUG]: can't debug unit tests due to 'code' name collision
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When debugging unit tests using PyCharm pytest template, session fails with error:
`ImportError: cannot import name 'InteractiveConsole' from 'code' (/Users/ericvergnaud/Development/databricks/ucx/tests/unit/code/__init__.py)`

### Expected Behavior

We should be able to debug unit tests

### Steps To Reproduce

Use PyCharm
Uncomment `test_runtime_delete_backup_groups` in `tests/unit/test_runtime.py` 
Try debugging via right-click

This issue is caused by our 'code' module which overrides the python 'code' module when running pytest.

### Cloud

AWS

### Operating System

macOS

### Version

latest via Databricks CLI

### Relevant log output

```shell
/Users/ericvergnaud/.pyenv/versions/3.12.2/bin/python3 -X pycache_prefix=/Users/ericvergnaud/Library/Caches/JetBrains/PyCharmCE2023.3/cpython-cache /Applications/PyCharm CE.app/Contents/plugins/python-ce/helpers/pydev/pydevd.py --multiprocess --qt-support=auto --client 127.0.0.1 --port 57247 --file /Applications/PyCharm CE.app/Contents/plugins/python-ce/helpers/pycharm/_jb_pytest_runner.py --target test_runtime.py::test_runtime_apply_permissions_to_account_groups 
Testing started at 17:04 ...
Traceback (most recent call last):
  File "/Applications/PyCharm CE.app/Contents/plugins/python-ce/helpers/pydev/_pydevd_bundle/pydevd_console_integration.py", line 4, in <module>
    from code import InteractiveConsole, InteractiveInterpreter
ImportError: cannot import name 'InteractiveConsole' from 'code' (/Users/ericvergnaud/Development/databricks/ucx/tests/unit/code/__init__.py)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Applications/PyCharm CE.app/Contents/plugins/python-ce/helpers/pydev/pydevd.py", line 49, in <module>
    from _pydevd_bundle.pydevd_comm import CMD_SET_BREAK, CMD_SET_NEXT_STATEMENT, CMD_STEP_INTO, CMD_STEP_OVER, \
  File "/Applications/PyCharm CE.app/Contents/plugins/python-ce/helpers/pydev/_pydevd_bundle/pydevd_comm.py", line 91, in <module>
    from _pydevd_bundle import pydevd_console_integration
  File "/Applications/PyCharm CE.app/Contents/plugins/python-ce/helpers/pydev/_pydevd_bundle/pydevd_console_integration.py", line 6, in <module>
    from _pydevd_bundle.pydevconsole_code_for_ironpython import IronPythonInteractiveConsole as InteractiveConsole, \
  File "/Applications/PyCharm CE.app/Contents/plugins/python-ce/helpers/pydev/_pydevd_bundle/pydevconsole_code_for_ironpython.py", line 305
    exec code in self.locals
    ^^^^^^^^^^^^^^^^^^^^^^^^
SyntaxError: Missing parentheses in call to 'exec'. Did you mean exec(...)?

Process finished with exit code 1

Empty suite
```

### PR DESCRIPTION
**Pull request by ericvergnaud**
## Changes
Renames 'code' module to 'source_code'

### Linked issues
Resolves #1126 

### Functionality 

- [ ] added relevant user documentation
- [ ] added new CLI command
- [ ] modified existing command: `databricks labs ucx ...`
- [ ] added a new workflow
- [ ] modified existing workflow: `...`
- [ ] added a new table
- [ ] modified existing table: `...`

### Tests

- [x] manually tested
- [ ] added unit tests
- [ ] added integration tests
- [ ] verified on staging environment (screenshot attached)

Tested by running existing unit tests

### CODE REVIEW

File: tests/unit/test_runtime.py [hunk 1]
```diff
@@ -90,10 +105,143 @@ def test_runtime_grants(mocker):
         sql_backend = MockBackend()
         crawl_grants(cfg, ws, sql_backend, mock_installation())
 
-        assert "SHOW DATABASES FROM hive_metastore" in sql_backend.queries
+        assert "SELECT * FROM hive_metastore.ucx.grants" in sql_backend.queries
+
+
+# TODO crawl_permissions fails, but I don't think I can create tickets yet!
```
Comment (nfx) [inline tests/unit/test_runtime.py:111]: it seems the changes with commented code from the other PR leaked to this one. 

let's remove the non-rename code from this PR.

**Review by nfx — CHANGES_REQUESTED** [review]

File: tests/unit/test_runtime.py (hunk 1 above)
Comment (ericvergnaud) [reply tests/unit/test_runtime.py:111]: Yes the reason for this was to ensure that it fixed the problem encountered in the other PR.
I'll rebase.

Comment (ericvergnaud) [reply tests/unit/test_runtime.py:111]: (That said if you approve the other PR the resulting change should be equivalent)

Comment (nfx) [reply tests/unit/test_runtime.py:111]: let's remove the commented-out code :) it's always bad.

**Comment by ericvergnaud** [comment]
@nfx looks good to go

**Review by nfx — APPROVED** [review]

**Comment by ericvergnaud** [comment]
CI results are misleading here since all reported issues are pre-existing i. The only thing this PR does is change a module name and update corresponding imports...

**Pull request merged**